### PR TITLE
Streamline onboarding: Remove the pre-launch checklist

### LINKED ISSUE #2492: Onboarding: Remove Pre-Launch checklist UI
Part of https://github.com/woocommerce/google-listings-and-ads/issues/2458

The 3rd step of the onboarding process includes this pre-launch check list at the bottom of the screen.

Image

This checklist doesn't really add value to the user experience and can lead to someone either abandoning the onboarding process or just checking everything in order to move on. Let's remove this section to streamline the onboarding experience.

## Acceptance Criteria

- [ ] On the 3rd step of onboarding, "Confirm store requirements", the pre-launch section is no longer shown
- [ ] The `PreLaunchChecklist` component is removed from the codebase (js/src/setup-mc/setup-stepper/store-requirements/pre-launch-checklist/index.js)
- [ ] Options related to the pre launch checklist should no longer be updated during this onboarding step once the `PreLaunchChecklist` is removed.

## Implementation Brief
The `PreLaunchChecklist` component is defined in `js/src/setup-mc/setup-stepper/store-requirements/pre-launch-checklist/index.js` and imported into the `StoreRequirements` page in `js/src/setup-mc/setup-stepper/store-requirements/index.js`. We can likely just remove the `PreLaunchChecklist` component from the the `StoreRequirements` page and delete the file where it's defined, since it doesn't seem to be used anywhere else. This `useEffect()` hook ([link](https://github.com/woocommerce/google-listings-and-ads/blob/a8e9cac5cace2cb24efb076427308d1b16ce69cf/js/src/setup-mc/setup-stepper/store-requirements/index.js#L82-L121)) that updates the saved settings based on `policyCheckData` can also be removed.

## Test Coverage

- Update E2E tests in tests/e2e/specs/setup-mc/step-3-confirm-store-requirements.test.js to remove tests related to the Pre-Launch Checklist.

## Definition Questions

1. It looks like these checkboxes do actually trigger an HTTP request to the `/wp-json/wc/gla/mc/settings` endpoint, which ends up updating the DB option stored in `gla_merchant_center` but these values don't persist when the page is reloaded. Is there any other JS code that needs to be cleaned up when this is removed so the onboarding process doesn't affect the stored DB values if previously set?
2. The JEST tests for the pre-launch-checker would be removed. Are there other JEST tests or storybook snapshots that we need to plan to update?

### PR DESCRIPTION
**Pull request by asvinb**
### Changes proposed in this Pull Request:

Closes #2492  .
* Removed the `PreLaunchChecklist` component from the codebase, including any associated tests.
* [Removed the check ](https://github.com/woocommerce/google-listings-and-ads/pull/2532/files#diff-e16c9ef1afbb70e7498ad66eb8bbbca6c3f7cb87f78a7c5e4309234989602639L172)if the form is valid or not before continuing because there are no more checkboxes.
* Disabled the `Continue` button in case the phone number is not verified or store address not filled in.
* Updated E2E tests to account for the previous point.

### Screenshots:
image

### Detailed test instructions:

1. Go through the onboarding process.
2. On the third step, i.e `Confirm store requirements`, the Pre Launch Checklist section should no longer be present.
3. Ensure the user can proceed with the next steps in the onboarding process.

### Changelog entry

> Update - Remove the Pre-Launch Checklist from the onboarding flow.

### CODE REVIEW

File: js/src/setup-mc/setup-stepper/store-requirements/index.js
Comment (joemcgill) [inline js/src/setup-mc/setup-stepper/store-requirements/index.js]: This might be a good idea, but I think it's unnecessary. Currently, the form can be submitted as long as the settings aren't being saved. Once saved, if there is some validation that fails, the `handleSubmitClick` callback will show validation errors rather than moving on to the next step. Therefore, I only think we need to disable this button if we're in the middle of recording some state that would change the validation (not something that is currently applicable).

**Review by joemcgill — CHANGES_REQUESTED** [review]
Thanks @asvinb, this is looking pretty good. I left feedback on the disabled button change inline. Additionally, there is one other E2E test that needs to be updated [here](https://github.com/woocommerce/google-listings-and-ads/blob/feature/2492-remove-pre-launch-checklist/tests/e2e/specs/setup-mc/step-2-product-listings.test.js#L433-L438). This test from Step 2 is confirming that the REST requests for the policy checks are being made after moving on from step 2. Now that those are removed, these can also be removed from that test.

**Review by joemcgill — APPROVED** [review]
Updates look good. Thanks @asvinb!

**Comment by ankitguptaindia** [comment]
Hi @asvinb, thanks for the PR. I tested this PR and found it to be working as expected. The Pre-Launch Checklist has been removed from the onboarding flow ✅.


>The "Continue" button should be disabled if the phone number has not been verified or the store address not filled in.

1. I have some concerns related to the status of the "Continue" button. ⚠️ 

Please take a look at the use cases below and let me know if they are not related to the changes completed in this PR **_and if they are worth addressing in another PR/issue:_**

I noticed that when the store doesn’t have a verified phone number, the "Continue" button is active, even though it doesn’t allow you to move to the next step. This can be confusing. Please see the attached screenshots.

I observed inconsistent behavior in the onboarding flow. On the first step, when required information or accounts are not added/connected, the button remains inactive (works as expected). However, on the second screen, when required details like 'Shipping times' are not added, the "Continue" button remains active, as it does on the third screen. So on 1st screen, continue button appears inactive and on 2nd, 3rd screen it appear as active when required details are not added.  

2. If a merchant already has a verified phone number and changes the number on the third step, it sends a verification code to verify the new number. However, if the merchant clicks the "Continue" button, the onboarding flow completes based on the previous number. 

<Details>
  <summary>Please look at below screenshots/screemcast</summary>
  
  

https://github.com/user-attachments/assets/40899613-cf1f-4290-82e1-590b3c21b446

First screen: 

image


Second/third screens:

image


image

Screenshot 2024-08-22 at 5 20 38 PM

</Details>


cc: @joemcgill for progress update.

**Comment by asvinb** [comment]
@ankitguptaindia Thanks for your comment. Apologies, I did not update the QAB since disabling the continue button was something which I added. You can see @joemcgill 's [comment here](https://github.com/woocommerce/google-listings-and-ads/pull/2532#discussion_r1720094627) about it. Let me know if that answers your questions.

**Comment by ankitguptaindia** [comment]
**QA/Test Report-** ✅ 
------------

**Testing Environment -** 

<Details>

* WordPress: 6.6.1
* Theme active on store: Storefront Version: 4.6.0
* WooCommerce - Version 9.2.1 
* PHP: 8.3
* Web Server: Nginx
* Browser: Chrome - Version 127
* OS: macOS Sonoma 14.6.1 

</Details>


**Test Results -**  Changes working as expected.  PreLaunchChecklist component has been removed and the onboarding flow working as expected. 

**Next Step-** Ready to Code Review(Woo) 


**Plugin file with this fix-specific code:**  [google-listings-and-ads.zip](https://github.com/user-attachments/files/16710690/google-listings-and-ads.zip)

**Review by ankitguptaindia — APPROVED** [review]

**Comment by joemcgill** [comment]
@eason9487 this one is ready for your review

File: tests/e2e/specs/setup-mc/step-2-product-listings.test.js
Comment (jorgemd24) [inline tests/e2e/specs/setup-mc/step-2-product-listings.test.js:407]: I believe the `/gla/mc/contact-information` request is still being fetched, so instead of asserting the policy check response, could we assert the contact information instead? Otherwise, we're just asserting the heading without checking any of the requests. Also, the title could be updated to align with these changes.

File: tests/e2e/specs/setup-mc/step-3-confirm-store-requirements.test.js
Comment (jorgemd24) [inline tests/e2e/specs/setup-mc/step-3-confirm-store-requirements.test.js:84]: Thanks for adjusting this.

**Review by jorgemd24 — COMMENTED** [review]
Hi @asvinb, it's working well, just a few comments:

This hook can be removed: https://github.com/woocommerce/google-listings-and-ads/issues/2492#issuecomment-2270939796 since it's no longer used in the codebase.

I know this PR focuses on the JS part, but what about the backend code used for the pre-launch checklist? For instance, `src/API/Site/Controllers/MerchantCenter/PolicyComplianceCheckController.php`—is it still needed, or will it be handled in a different PR?

Also, there are other parts of the JS code that still reference `policy_check`. Should these be kept, or can they be removed as well?

https://github.com/woocommerce/google-listings-and-ads/blob/32bbb301cbf9e517ccf9b45a487359d328239911/js/src/data/reducer.js#L419-L422

https://github.com/woocommerce/google-listings-and-ads/blob/32bbb301cbf9e517ccf9b45a487359d328239911/js/src/data/resolvers.js#L174-L195

File: tests/e2e/specs/setup-mc/step-2-product-listings.test.js
Comment (asvinb) [reply tests/e2e/specs/setup-mc/step-2-product-listings.test.js:407]: Sounds good @jorgemd24 . I've now updated the PR for additional tests.

**Comment by asvinb** [comment]
Hello @jorgemd24 ,
Thanks for the review. Removing the PHP part is tackled in the following ticket: https://github.com/woocommerce/google-listings-and-ads/issues/2494 
I've also removed any occurrences of policy_check :)

**Review by jorgemd24 — APPROVED** [review]
Thanks for the adjustments, LGTM. I couldn’t find any more references to the policy check in the frontend, and it’s working well.

**Comment by joemcgill** [comment]
Thanks all. @asvinb feel free to merge this after resolving conflicts and then move to UAT.

**Comment by asvinb** [comment]
Thanks @joemcgill @jorgemd24 . Merge conflicts fixed now.

**Comment by eason9487** [comment]
Hi @asvinb, please hold off on the merge. ~There is an issue needs to be fixed.~ There are a few leftover codes need to be cleared together.

File: js/src/setup-mc/setup-stepper/store-requirements/index.js
Comment (eason9487) [inline js/src/setup-mc/setup-stepper/store-requirements/index.js]: The `extendAdapter` can be removed.

**Review by eason9487 — COMMENTED** [review]
Thanks for the work!

It would be appreciated if the following unused codes could be removed together:

- The `extendAdapter` prop of `AdaptiveForm` in the js/src/setup-mc/setup-stepper/store-requirements/index.js as [its only use](https://github.com/woocommerce/google-listings-and-ads/blob/1030b8d67b129c4206548c3f3cefb854c473fff5/js/src/setup-mc/setup-stepper/store-requirements/pre-launch-checklist/index.js#L186) will be removed by this PR
- All `PreLaunchCheckItem` component codes under [this directory](https://github.com/woocommerce/google-listings-and-ads/tree/c21b1b5e5d4ddcdb7aa857042e6d01255676f421/js/src/components/pre-launch-check-item)
- E2E utils:
   - https://github.com/woocommerce/google-listings-and-ads/blob/c21b1b5e5d4ddcdb7aa857042e6d01255676f421/tests/e2e/utils/pages/setup-mc/step-3-confirm-store-requirements.js#L58-L65
   - https://github.com/woocommerce/google-listings-and-ads/blob/c21b1b5e5d4ddcdb7aa857042e6d01255676f421/tests/e2e/utils/pages/setup-mc/step-3-confirm-store-requirements.js#L170-L186
   - https://github.com/woocommerce/google-listings-and-ads/blob/c21b1b5e5d4ddcdb7aa857042e6d01255676f421/tests/e2e/utils/pages/setup-mc/step-3-confirm-store-requirements.js#L221-L231
   - https://github.com/woocommerce/google-listings-and-ads/blob/c21b1b5e5d4ddcdb7aa857042e6d01255676f421/tests/e2e/utils/pages/setup-mc/step-3-confirm-store-requirements.js#L245-L266
- All frontend codes and comments related to these keywords:
   - `website_live`
   - `checkout_process_secure`
   - `payment_methods_visible`
   - `refund_tos_visible`
   - `contact_info_visible`

File: js/src/setup-mc/setup-stepper/store-requirements/index.js
Comment (asvinb) [reply js/src/setup-mc/setup-stepper/store-requirements/index.js]: Thanks @eason9487 . Removed.

**Comment by asvinb** [comment]
@eason9487 Kindly note I've updated your PR following your last [comment](https://github.com/woocommerce/google-listings-and-ads/pull/2532#pullrequestreview-2276799240). Can you check again please?

**Review by eason9487 — APPROVED** [review]
Thanks for the further cleanup! LGTM.

**Pull request merged**
